pkg/wasm/imagefetcher.go

```diff
@@ -122,6 +135,22 @@ func (o *ImageFetcher) Fetch(url, expManifestDigest string) ([]byte, error) {
 	)
 }
 
+func parseReference(url string) (name.Reference, error) {
+	ref, err := name.ParseReference(url)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse url in image reference: %v", err)
```

Comment:
nit: could just return the error here, otherwise the error will be wrapped again with the same text after returning.